content/blog/building-original-projects/index.md

```diff
@@ -0,0 +1,285 @@
+
+The rest of the line, `c is not defined` refers to the `message`. This is the meaning of the error, and simply means that `c` does not exist as a variable in the current scope.
+
+The last three lines are the `stack trace`, a list of function calls that lead to the point where the error was thrown. In order to read this, we'll start at the last line:
+
+ ```
+ at script.js:13:1
+ ```
+
+This is saying that we began in the file `script.js`, at line 13, column 1. If we go to that line in our code, we see the print function is called like so:
+
+```javascript
+print();
+```
+
+From there we can move up the stack trace and read the next line:
+
+```
+at print (script.js:9:15)
+```
+
+If we look at line 9, column 15 in the same `script.js` file, we'll notice another function call, this time to `add`:
+
+```javascript
+const sum = add(a, b);
+```
+
+Now we can follow the stack trace up further:
+
+```at add (script.js:5:3)
+at add (script.js:5:3)
+```
+
+Which tells us that we should look on line 5:
+
+```
+function add(a, b) {
+  return c; // This looks like the culprit
+}
+```
+
+Bingo! It seems like this line, `return c;` is the issue, and true to the error's message, it is trying to reference a `c` variable that has not been defined in the scope of this function. What we wanted to do was store the addition of `a` and `b` in a `c` variable and return it, like so:
+
+``` 
+function add(a, b) {
+  const c = a + b;
+  return c;
+}
+```
+
+This solves the error and now if we run the script, `7` is logged to the console.
+
+### Researching error messages
+
+After you've looked into the message and stack trace, you might still have questions. In this case, it's okay to copy-and-paste the message into Google or StackOverflow. Someone else may have run into the same error and posted about it, or a very similar error that can still lead to insight. You might also check out the docs or forum for your technology. However, I'd still recommend parsing your message manually first, as doing so can give you useful information like the type of the error, which you can also look up and read about.
+
+> When to use: If your problem stems from a bug that produces an error message, always!
+
+## Try Debugging
+How you debug will depend on your choice of programming language and code editor, and there are even debuggers that work on the command line or in your browser. If you use Chrome, one way to practice is to use the built-in debugger, [as outlined in this official guide to debugging JavaScript in Chrome](https://developer.chrome.com/docs/devtools/javascript/). The basic premise of debuggers are that you can observe the behavior of your code as it's executing. By setting one or several `breakpoints`, your debugger will allow you to slowly step forward line-by-line starting from one of the breakpoints set. The benefit of this is that you can examine the state of your variables in the current scope as you move through the execution of your code.   
```

Comment:
Did you know that you can also use Chrome's debugger in Node? Maybe we could add a link to my article? :) 

https://unicorn-utterances.com/posts/debugging-nodejs-programs-using-chrome